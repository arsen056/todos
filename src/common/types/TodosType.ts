export type FilterType = 'all' | 'active' | 'completed'

export interface TodosType {
  title: string,
  filter: FilterType
  tasks: TaskType[]
}

export interface TaskType {
  id: string
  title: string,
  isDone: boolean
}