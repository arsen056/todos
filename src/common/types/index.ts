export * from 'common/types/TodosType';
export * from 'common/types/ContextType';

