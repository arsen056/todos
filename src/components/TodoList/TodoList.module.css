.todoList {
  background-color: #FFFFFF;
  width: 500px;
  margin: 2rem;
  box-shadow: 0 0 8px 0 rgba(34, 60, 80, 0.2);
  border-radius: 10px;
  padding: 1rem 2rem;
}

.title {
  font-size: 30px;
  padding: 0.5rem 2rem;
}