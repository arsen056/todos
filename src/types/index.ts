export * from './TaskType';
export * from './FilterType';
export * from './TodoType';
export * from './ContextData';

