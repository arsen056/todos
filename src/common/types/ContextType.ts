import {Dispatch} from 'react';
import {TodosType} from 'common/types/TodosType';

export interface ContextType {
  todos: TodosType,
  dispatch: Dispatch<any>
}