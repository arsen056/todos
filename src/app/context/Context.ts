import {createContext} from 'react';
import {ContextType} from 'types';

export const Context = createContext<ContextType>({} as ContextType);