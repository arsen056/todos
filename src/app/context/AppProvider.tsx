import React, {FC, ReactNode, useReducer} from 'react';
import {State, reducer} from 'reducer/reducer';
import {Context} from 'app/context/Context';
import {TodosType} from 'common/types';

interface Props {
  children: ReactNode
  initState?: TodosType
}

export const AppProvider:FC<Props> = ({children, initState}) => {
  const RootState = initState ? initState : State;

  const [todos, dispatch] = useReducer(reducer, RootState);

  return (
    <Context.Provider value={{todos, dispatch}}>
      {children}
    </Context.Provider>
  );
};
