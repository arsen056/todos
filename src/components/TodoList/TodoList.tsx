import React, {useContext} from 'react';

import {TaskType} from 'types';

import {Task} from 'components/Task/Task';

import {Context} from 'app/context/Context';

import {addTask} from 'reducer/reducer';

import s from './TodoList.module.css';

export const TodoList = () => {

  const {todos, dispatch} = useContext(Context);

  const tasksMap = todos.map(t => <Task title={t.title} isDone={t.isDone} />);

  const addNewTask = (title: string) => {
    const newTask: TaskType = {id: '1', title, isDone: false};
    dispatch(addTask(newTask));
  };

  return (
    <div className={s.todoList}>
      <h2>{'title'}</h2>
      <input type="text"/>
      <button onClick={() => addNewTask('hello')}>+</button>
      <ul>
        {tasksMap}
      </ul>
    </div>
  );
};