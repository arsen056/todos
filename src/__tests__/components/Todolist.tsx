import {render} from '@testing-library/react';
import {TodoList} from 'components/TodoList/TodoList';
import {AppProvider} from 'app/context/AppProvider';
import {TodosType} from 'common/types';
import uuid from 'react-uuid';

const State: TodosType = {
  title: 'Todos',
  filter: 'all',
  tasks: [
    {id: uuid(), title: 'React', isDone: false},
    {id: uuid(), title: 'Html', isDone: false},
    {id: uuid(), title: 'Css', isDone: false},
    {id: uuid(), title: 'JS', isDone: true}]
};

describe('Todolist', () => {
  it('Should be render todolist', () => {
    const view = render(
      <AppProvider initState={State}>
        <TodoList />
      </AppProvider>
    );    
    expect(view).toMatchSnapshot();
  });

  it('Should create todolist with empty todos', () => {
    const view = render(
      <AppProvider initState={{...State, tasks: []}}>
        <TodoList />
      </AppProvider>
    );

    expect(view).toMatchSnapshot();
  });

  it('Should create todolist with filter active', () => {
    const view = render(
      <AppProvider initState={{...State, filter: 'active'}}>
        <TodoList />
      </AppProvider>
    );
    expect(view).toMatchSnapshot();
  });

  it('Should create todolist with filter completed', () => {
    const view = render(
      <AppProvider initState={{...State, filter: 'completed'}}>
        <TodoList />
      </AppProvider>
    );
    expect(view).toMatchSnapshot();
  });
});
