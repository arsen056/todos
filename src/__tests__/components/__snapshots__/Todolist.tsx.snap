// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todolist Should be render todolist 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todoList"
      >
        <h2
          class="title"
        >
          Todos
        </h2>
        <div
          class="addItem"
        >
          <span
            class="inputWrapper"
          >
            <input
              class="input input  fullwidth"
              placeholder="Enter title"
              type="text"
              value=""
            />
          </span>
          <button
            class="button button"
          >
            +
          </button>
        </div>
        <div
          class="group"
        >
          <button
            class="button active"
          >
            all
          </button>
          <button
            class="button "
          >
            active
          </button>
          <button
            class="button "
          >
            completed
          </button>
          <button
            class="button undefined"
          >
            Clear
          </button>
        </div>
        <p
          class="itemsLeft"
        >
          3
           items left
        </p>
        <ul>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                React
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                Html
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                Css
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task checked"
              >
                <input
                  checked=""
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span checked"
              >
                JS
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todoList"
    >
      <h2
        class="title"
      >
        Todos
      </h2>
      <div
        class="addItem"
      >
        <span
          class="inputWrapper"
        >
          <input
            class="input input  fullwidth"
            placeholder="Enter title"
            type="text"
            value=""
          />
        </span>
        <button
          class="button button"
        >
          +
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="button active"
        >
          all
        </button>
        <button
          class="button "
        >
          active
        </button>
        <button
          class="button "
        >
          completed
        </button>
        <button
          class="button undefined"
        >
          Clear
        </button>
      </div>
      <p
        class="itemsLeft"
      >
        3
         items left
      </p>
      <ul>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              React
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              Html
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              Css
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task checked"
            >
              <input
                checked=""
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span checked"
            >
              JS
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Todolist Should create todolist with empty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todoList"
      >
        <h2
          class="title"
        >
          Todos
        </h2>
        <div
          class="addItem"
        >
          <span
            class="inputWrapper"
          >
            <input
              class="input input  fullwidth"
              placeholder="Enter title"
              type="text"
              value=""
            />
          </span>
          <button
            class="button button"
          >
            +
          </button>
        </div>
        <div
          class="group"
        >
          <button
            class="button active"
          >
            all
          </button>
          <button
            class="button "
          >
            active
          </button>
          <button
            class="button "
          >
            completed
          </button>
          <button
            class="button undefined"
          >
            Clear
          </button>
        </div>
        <p
          class="itemsLeft"
        >
          3
           items left
        </p>
        <ul>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                React
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                Html
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                Css
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task checked"
              >
                <input
                  checked=""
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span checked"
              >
                JS
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todoList"
    >
      <h2
        class="title"
      >
        Todos
      </h2>
      <div
        class="addItem"
      >
        <span
          class="inputWrapper"
        >
          <input
            class="input input  fullwidth"
            placeholder="Enter title"
            type="text"
            value=""
          />
        </span>
        <button
          class="button button"
        >
          +
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="button active"
        >
          all
        </button>
        <button
          class="button "
        >
          active
        </button>
        <button
          class="button "
        >
          completed
        </button>
        <button
          class="button undefined"
        >
          Clear
        </button>
      </div>
      <p
        class="itemsLeft"
      >
        3
         items left
      </p>
      <ul>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              React
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              Html
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              Css
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task checked"
            >
              <input
                checked=""
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span checked"
            >
              JS
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Todolist Should create todolist with empty todos 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todoList"
      >
        <h2
          class="title"
        >
          Todos
        </h2>
        <div
          class="addItem"
        >
          <span
            class="inputWrapper"
          >
            <input
              class="input input  fullwidth"
              placeholder="Enter title"
              type="text"
              value=""
            />
          </span>
          <button
            class="button button"
          >
            +
          </button>
        </div>
        <div
          class="group"
        >
          <button
            class="button active"
          >
            all
          </button>
          <button
            class="button "
          >
            active
          </button>
          <button
            class="button "
          >
            completed
          </button>
          <button
            class="button undefined"
          >
            Clear
          </button>
        </div>
        <p
          class="itemsLeft"
        >
          all done
        </p>
        <ul>
          <h3
            class="empty"
          >
            List empty...
          </h3>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todoList"
    >
      <h2
        class="title"
      >
        Todos
      </h2>
      <div
        class="addItem"
      >
        <span
          class="inputWrapper"
        >
          <input
            class="input input  fullwidth"
            placeholder="Enter title"
            type="text"
            value=""
          />
        </span>
        <button
          class="button button"
        >
          +
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="button active"
        >
          all
        </button>
        <button
          class="button "
        >
          active
        </button>
        <button
          class="button "
        >
          completed
        </button>
        <button
          class="button undefined"
        >
          Clear
        </button>
      </div>
      <p
        class="itemsLeft"
      >
        all done
      </p>
      <ul>
        <h3
          class="empty"
        >
          List empty...
        </h3>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Todolist Should create todolist with filter active 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todoList"
      >
        <h2
          class="title"
        >
          Todos
        </h2>
        <div
          class="addItem"
        >
          <span
            class="inputWrapper"
          >
            <input
              class="input input  fullwidth"
              placeholder="Enter title"
              type="text"
              value=""
            />
          </span>
          <button
            class="button button"
          >
            +
          </button>
        </div>
        <div
          class="group"
        >
          <button
            class="button active"
          >
            all
          </button>
          <button
            class="button "
          >
            active
          </button>
          <button
            class="button "
          >
            completed
          </button>
          <button
            class="button undefined"
          >
            Clear
          </button>
        </div>
        <p
          class="itemsLeft"
        >
          3
           items left
        </p>
        <ul>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                React
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                Html
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task "
              >
                <input
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span "
              >
                Css
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todoList"
    >
      <h2
        class="title"
      >
        Todos
      </h2>
      <div
        class="addItem"
      >
        <span
          class="inputWrapper"
        >
          <input
            class="input input  fullwidth"
            placeholder="Enter title"
            type="text"
            value=""
          />
        </span>
        <button
          class="button button"
        >
          +
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="button active"
        >
          all
        </button>
        <button
          class="button "
        >
          active
        </button>
        <button
          class="button "
        >
          completed
        </button>
        <button
          class="button undefined"
        >
          Clear
        </button>
      </div>
      <p
        class="itemsLeft"
      >
        3
         items left
      </p>
      <ul>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              React
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              Html
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task "
            >
              <input
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span "
            >
              Css
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Todolist Should create todolist with filter completed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todoList"
      >
        <h2
          class="title"
        >
          Todos
        </h2>
        <div
          class="addItem"
        >
          <span
            class="inputWrapper"
          >
            <input
              class="input input  fullwidth"
              placeholder="Enter title"
              type="text"
              value=""
            />
          </span>
          <button
            class="button button"
          >
            +
          </button>
        </div>
        <div
          class="group"
        >
          <button
            class="button active"
          >
            all
          </button>
          <button
            class="button "
          >
            active
          </button>
          <button
            class="button "
          >
            completed
          </button>
          <button
            class="button undefined"
          >
            Clear
          </button>
        </div>
        <p
          class="itemsLeft"
        >
          3
           items left
        </p>
        <ul>
          <li
            class="taskWrapper"
          >
            <div
              class="taskItem"
            >
              <label
                class="task checked"
              >
                <input
                  checked=""
                  class="checkbox"
                  type="checkbox"
                />
                <span
                  class="customCheckbox"
                />
              </label>
              <span
                class="span checked"
              >
                JS
              </span>
            </div>
            <button
              class="button undefined"
            >
              x
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todoList"
    >
      <h2
        class="title"
      >
        Todos
      </h2>
      <div
        class="addItem"
      >
        <span
          class="inputWrapper"
        >
          <input
            class="input input  fullwidth"
            placeholder="Enter title"
            type="text"
            value=""
          />
        </span>
        <button
          class="button button"
        >
          +
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="button active"
        >
          all
        </button>
        <button
          class="button "
        >
          active
        </button>
        <button
          class="button "
        >
          completed
        </button>
        <button
          class="button undefined"
        >
          Clear
        </button>
      </div>
      <p
        class="itemsLeft"
      >
        3
         items left
      </p>
      <ul>
        <li
          class="taskWrapper"
        >
          <div
            class="taskItem"
          >
            <label
              class="task checked"
            >
              <input
                checked=""
                class="checkbox"
                type="checkbox"
              />
              <span
                class="customCheckbox"
              />
            </label>
            <span
              class="span checked"
            >
              JS
            </span>
          </div>
          <button
            class="button undefined"
          >
            x
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
